recipes/libssh/all/test_package/test_package.c

```diff
@@ -1,6 +1,14 @@
 #include <libssh/libssh.h>
 
 int main(int argc, char *argv[]) {
+  ssh_init();
   ssh_session session = ssh_new();
+  if (session == NULL)
+    return -1;
+
+  const char* host = "localhost";
+  ssh_options_set(session, SSH_OPTIONS_HOST, host);
+  ssh_options_set(session, SSH_OPTIONS_USER, "user");
+  int rc = ssh_connect(session);
```

Comment:
Can we go back to a simpler test package? We don't like having connetcions that interact with the network layers whre possible, that's why it was so simple in the first place.

Did this catch any issues in your testing? I'd like to also get some more insight if it turned out to have been useful while developing the recipe :)

Reply:
Yeah, it caught issues in the testing. Below are some notes on the build with mbedtls/*:enable_threading=False (one of the failing cases). I'll push a commit up soon to simplify it back down

Without the call to `ssh_connect` the test package exits successfully, even though there is a log line that indicates there is something wrong

```
# ./recipes/libssh/all/test_package/build/gcc-11-x86_64-gnu17-debug/test_package
Error in auto_init()
# echo #?
0
```

With the call to `ssh_connect` the test package core dumps, which also causes the `conan create` command to fail. This was useful while developing the recipe, as I was able to attach gdb to the test_package and step through to find where exactly the issue was

```
# ./recipes/libssh/all/test_package/build/gcc-11-x86_64-gnu17-debug/test_package
Error in auto_init()
[1]    3151354 segmentation fault (core dumped)
# echo $?
139


# conan create recipes/libssh/all --version 0.10.6 -o crypto_backend=mbedtls -o "mbedtls/*:enable_threading=False" -s "build_type=Debug" -u -b missing
[...]
======== Testing the package: Executing test ========
libssh/0.10.6 (test package): Running test()
libssh/0.10.6 (test package): RUN: ./test_package
Error in auto_init()

ERROR: libssh/0.10.6 (test package): Error in test() method, line 26
        self.run(bin_path, env="conanrun")
        ConanException: Error -11 while executing
```

Reply:
This is super appreciated! If there was a way to catch issues like this in the future that did not need to actually perform the connection, I would be suuper open to have a stronger test